proxy/sources/rawkuma.py

```diff
@@ -0,0 +1,197 @@
+            try:
+                for element in soup.find_all("div", class_="thumb")[0].children:
+                    if element.name == 'img':
+                        cover = 'https://' + re.sub(r'(https?:)?\/\/', '', element['src'])
+            except AttributeError:
+                cover = ""
+
+            groups_dict = {"1": "N/A"}
+
+            chapter_list = []
+            chapter_dict = {}
+            chapters = list(
+                map(
+                    lambda a: [
+                        a.text,
+                        a.parent["href"],
+                        a.next_sibling.next_sibling.text
+                    ],
+                    soup.find_all("span", class_="chapternum")
+                )
+            )
+            for chapter in chapters:
+                canonical_chapter = self.parse_chapter(chapter[1])
+                chapter_list.append(
+                    [
+                        "",
+                        canonical_chapter,
+                        chapter[0],
+                        canonical_chapter.replace(".", "-"),
+                        "N/A",
+                        chapter[2],
+                        "",
+                    ]
+                )
+                chapter_dict[canonical_chapter] = {
+                    "volume": "1",
+                    "title": chapter[0],
+                    "groups": { "1": self.wrap_chapter_meta(self.get_slug_name(chapter[1])) },
+                }
+
+            return {
+                "slug": meta_id,
+                "title": title,
+                "description": description,
+                "series": title,
+                "alt_titles_str": None,
+                "cover_vol_url": cover,
+                "metadata": [],
+                "author": author,
+                "artist": artist,
+                "groups": groups_dict,
+                "cover": cover,
+                "chapter_dict": chapter_dict,
+                "chapter_list": chapter_list,
+            }
+        else:
+            return None
+
+    @api_cache(prefix="rk_series_dt", time=600)
+    def series_api_handler(self, meta_id):
+        data = self.rk_scrape_common(meta_id)
+        if data:
+            return SeriesAPI(
+                slug=data["slug"],
+                title=data["title"],
+                description=data["description"],
+                author=data["author"],
+                artist=data["artist"],
+                groups=data["groups"],
+                cover=data["cover"],
+                chapters=data["chapter_dict"],
+            )
+        else:
+            return None
+
+    @api_cache(prefix="rk_chapter_dt", time=3600)
+    def chapter_api_handler(self, chapter_meta_id):
+        chapter_url = "https://rawkuma.com/" + chapter_meta_id
+        resp = get_wrapper(chapter_url)
+        if resp.status_code == 200:
+            data = resp.text
+            r = re.compile(r'"images"\s?:\s?\[[^]]*\]')
+            m = re.search(r, data)
```

Comment:
ooc: any reason you chose to compile the regexes here but not in the other methods where you used regexes?